cpp/src/io/json/json_gpu.cu

```diff
@@ -41,56 +43,53 @@ using string_pair = std::pair<const char *, size_t>;
 
 namespace {
 /**
- * @brief CUDA Kernel that modifies the start and stop offsets to exclude
- * the sections outside of the top level brackets.
+ * @brief CUDA Kernel that adjusts the row range to exclude the character outside of the top level
+ * brackets.
  *
  * The top level brackets characters are excluded from the resulting range.
- * Parameter stop has the same semantics as end() in STL containers
- * (one past the last element)
  *
- * @param[in] data Pointer to the device buffer containing the data to process
- * @param[in,out] start Offset of the first character in the range
- * @param[in,out] stop Offset of the first character after the range
- *
- * @return void
- **/
-__device__ void limit_range_to_brackets(const char *data, long &start, long &stop)
+ * @param[in] begin Pointer to the first character in the row
+ * @param[in] end pointer to the first character after the row
+ */
+__device__ std::pair<char const *, char const *> limit_range_to_brackets(char const *begin,
+                                                                         char const *end)
 {
-  while (start < stop && data[start] != '[' && data[start] != '{') { start++; }
-  start++;
+  while (begin < end && *begin != '[' && *begin != '{') { begin++; }
+  begin++;
 
-  while (start < stop && data[stop - 1] != ']' && data[stop - 1] != '}') { stop--; }
-  stop--;
+  while (begin < end && *(end - 1) != ']' && *(end - 1) != '}') { end--; }
+  end--;
+  return {begin, end};
```

Comment:
Looks like (haven't tested but it compiled) you can use `thrust::make_reverse_iterator` with `char const *`, so we can update code to something like:
```cpp
  begin = thrust::find_if(thrust::device, begin, end, [](auto c) { return c == '[' || c == '{'; });
  end   = thrust::find_if(thrust::device,
                        thrust::make_reverse_iterator(end),
                        thrust::make_reverse_iterator(++begin),
                        [](auto c) { return c == ']' || c == '}'; })
          .base();
  return {begin, --end};
```